asoh/models/ecm/__init__.py

```diff
@@ -11,46 +11,19 @@
 
 
 class EquivalentCircuitModel(CellModel):
-    """Equivalent Circuit Model (ECM) representation of a battery
-
-    Args:
-        use_series_capacitor
-            Boolean to determine whether to employ a series capacitor.
-        number_RC_components
-            Number of RC components of equivalent circuit. Must be non-negative.
-        ASOH
-            Advanced State of Health (A-SOH) of the system. Used to parametrize
-            the dynamics of the system. It does not need to be provided on
-            initialization, but, if that is the case, it must be set on
-            subsequent function calls.
-        current_behavior
-            Determines how to the total current behaves in-between time steps.
-            Can be either 'constant' or 'linear'.
     """
+    Equivalent Circuit Model (ECM) dynamics of a battery
+    """
+
+    def __init__(self) -> None:
+        pass
 
-    def __init__(self,
-                 use_series_capacitor: bool = False,
-                 number_RC_components: int = 0,
-                 transient: ECMTransientVector = None,
-                 initial_input: ECMInput = None,
-                 current_behavior: Literal['constant', 'linear'] = 'constant'
-                 ) -> None:
-        self.num_C0 = int(use_series_capacitor)
-        self.num_RC = number_RC_components
-        self.current_behavior = current_behavior
-        # Lenght of hidden vector: SOC + q0 + I_RC_j + hysteresis
-        self.len_hidden = int(1 + self.num_C0 + self.num_RC + 1)
-        self.transient = transient
-        if initial_input is None:
-            initial_input = ECMInput(time=0., current=0.)
-        self.previous_input = initial_input
-
-    def update_transient_state(self,
-                               new_input: ECMInput,
-                               transient_state: Union[ECMTransientVector, None] = None,
-                               asoh: Union[ECMASOH, None] = None,
-                               previous_input: Union[ECMInput, None] = None,
-                               *args, **kwargs
+    @staticmethod
+    def update_transient_state(new_input: ECMInput,
+                               transient_state: ECMTransientVector,
+                               asoh: ECMASOH,
+                               previous_input: ECMInput,
+                               current_behavior: Literal['constant', 'linear'] = 'constant'
```

Comment:
I think `current_behavior` should remain an attribute of the simulator, rather than the model. This way, this class can continue being static in nature, and its main functionality can be used from its `staticmethod`s without needing separate instantiations.

Reply:
That's a good point. Static keeps us honest in designing the model to be stateless

Reply:
Ok, glad we agree. Marking as resolved for now.